src/App.js

```diff
@@ -1,86 +1,183 @@
-import React from 'react';
+import React, { useState, useMemo } from 'react';
+import { InputField } from './components/InputField';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { FILTERS } from './constants';
+
+const useLocalStorage = (key, initialValue) => {
+  const [value, setValue] = useState(() => {
+    try {
+      return JSON.parse(localStorage.getItem(key)) || initialValue;
+    } catch {
+      return initialValue;
+    }
+  });
+
+  const saveValue = (newValue) => {
+    setValue(newValue);
+    localStorage.setItem(key, JSON.stringify(newValue));
+  };
+
+  return [value, saveValue];
+};
```

Comment:
let's create a folder hooks and move this hook there (name file the same as a hook)